app/src/templates/user_profile.html

```diff
@@ -1,27 +1,55 @@
-{% extends "base.html" %}
-{% block title %}User Profile - Gamified Learning Platform{% endblock %}
-
+{% extends "base.html" %} {% block title %}User Profile - Gamified Learning Platform{% endblock %}
 {% block styling %}
-    <!-- Local Stylesheets -->
-    <link rel="stylesheet" href="{{ url_for('static', filename='css/register.css') }}">
-{% endblock %}
+<!-- Local Stylesheets -->
+<link rel="stylesheet" href="{{ url_for('static', filename='css/register.css') }}" />
+{% endblock %} {% block scripts %}
+<!-- jQuery -->
+<script src="https://code.jquery.com/jquery-3.6.4.min.js"></script>
+
+<script>
+    // User stats
+    function userStats() {
+        $.ajax({
+            url: "/user_stats",
+            method: "GET",
+            success: function (data) {
+                $("#streakDisplay").text(data.streak);
+                $("#pointsDisplay").text(data.points);
+            },
+            error: function (error) {
+                console.error("Error fetching user stats:", error);
+            },
+        });
+    }
+
+    $(document).ready(function () {
+        userStats();
+
+        // Update every 10 seconds
+        setInterval(userStats, 10000);
```

Comment:
This could cause an issue when we eventually migrate to cloud, so try to change this to 30s or even 1 minute